apiserver/facades/client/modelmanager/services.go

```diff
@@ -108,6 +109,18 @@ type ModelDefaultsService interface {
 	ModelDefaultsProvider(
 		uuid coremodel.UUID,
 	) modeldefaultsservice.ModelDefaultsProviderFunc
+
+	// CloudDefaults returns the default attribute details for a specified cloud.
+	// It returns an error satisfying [clouderrors.NotFound] if the cloud doesn't exist.
+	CloudDefaults(ctx context.Context, cloudName string) (modeldefaults.ModelDefaultAttributes, error)
+
+	// UpdateModelConfigDefaultValues saves the specified default attribute details for a cloud or region.
+	// It returns an error satisfying [clouderrors.NotFound] if the cloud doesn't exist.
+	UpdateModelConfigDefaultValues(ctx context.Context, updateAttrs map[string]interface{}, cloudRegion modeldefaults.CloudRegion) error
```

Comment:
question: How do you set the default attributes for the cloud? Only region is passed in as an arg. (See comment on struct naming)

question: What error is provided if the region doesn't exist?

Same for RemoveModelConfigDefaultValues

Aren't errors for the implementation of the interface, not the interface itself?

Reply:
The new implementation matches 1:1 the current semantics, which is also reflected in the CLI behaviour.
You supply just a cloud, you set the cloud values. You supply a region as well, you set the region values.
We're not attempting to re-engineer the underlying behaviour, just port across to dqlite.
If the region doesn't exist, you get a generic not found error. There's little value right now in doing anything else as the error is just emitted back to the caller and would need to be converted to a generic not found error if it were not that to start with.